internal/venice-client-common/src/main/java/com/linkedin/venice/stats/VeniceMetricsConfig.java

```diff
@@ -0,0 +1,235 @@
+      return this;
+    }
+
+    public VeniceMetricsConfigBuilder setMetricPrefix(String metricPrefix) {
+      this.metricPrefix = metricPrefix;
+      return this;
+    }
+
+    public VeniceMetricsConfigBuilder setEmitOpenTelemetryMetrics(boolean emitOpenTelemetryMetrics) {
+      this.emitOpenTelemetryMetrics = emitOpenTelemetryMetrics;
+      return this;
+    }
+
+    public VeniceMetricsConfigBuilder setEmitTehutiMetrics(boolean emitTehutiMetrics) {
+      this.emitTehutiMetrics = emitTehutiMetrics;
+      return this;
+    }
+
+    public VeniceMetricsConfigBuilder setEmitToHttpGrpcEndpoint(boolean emitToHttpGrpcEndpoint) {
+      this.emitToHttpGrpcEndpoint = emitToHttpGrpcEndpoint;
+      return this;
+    }
+
+    public VeniceMetricsConfigBuilder setEmitToLog(boolean emitToLog) {
+      this.emitToLog = emitToLog;
+      return this;
+    }
+
+    public VeniceMetricsConfigBuilder setMetricFormat(VeniceOpenTelemetryMetricFormat metricFormat) {
+      this.metricFormat = metricFormat;
+      return this;
+    }
+
+    public VeniceMetricsConfigBuilder extractAndSetOtelConfigs(Map<String, String> configs) {
+      // copy only OpenTelemetry related configs
+      for (Map.Entry<String, String> entry: configs.entrySet()) {
+        if (entry.getKey().startsWith("otel.")) {
+          otelConfigs.put(entry.getKey(), entry.getValue());
+        }
+      }
+      LOGGER.info("OpenTelemetry configs: {}", otelConfigs);
+      return this;
+    }
+
+    public VeniceMetricsConfigBuilder setTehutiMetricConfig(MetricConfig tehutiMetricConfig) {
+      this.tehutiMetricConfig = tehutiMetricConfig;
+      return this;
+    }
+
+    /** get the last part of the service name
+     * For instance: if service name is "venice-router", return "router"
+     */
+    public static String getMetricsPrefix(String input) {
+      String[] parts = input.split("[\\-\\._]");
+      String lastPart = parts[parts.length - 1];
+      return lastPart;
+    }
+
+    // Validate required fields before building
+    private void checkAndSetDefaults() {
+      if (tehutiMetricConfig == null) {
+        setTehutiMetricConfig(new MetricConfig());
+      }
+      if (metricPrefix == null) {
+        setMetricPrefix(getMetricsPrefix(serviceName));
+      }
+      if (otelConfigs.containsKey("otel.venice.enabled")) {
+        String status = otelConfigs.get("otel.venice.enabled");
+        if (status != null) {
+          setEmitOpenTelemetryMetrics(status.toLowerCase(Locale.ROOT).equals("true"));
+        }
+      }
+      // check otelConfigs and set defaults
+      if (emitOpenTelemetryMetrics) {
+        if (otelConfigs.containsKey("otel.venice.export.to.log")) {
+          String emitStatus = otelConfigs.get("otel.venice.export.to.log");
+          if (emitStatus != null) {
+            setEmitToLog(emitStatus.toLowerCase(Locale.ROOT).equals("true"));
+          }
+        }
+        if (otelConfigs.containsKey("otel.venice.export.to.http.grpc.endpoint")) {
```

Comment:
As discussed offline, because the `OtlpMetricExporterProvider` is marked internal, went with 
1. parsing the Venice Otel and Otel configs in venice
2. using the setter approach based on the parsed configs to create exporters.